src/client/datascience/dataScienceSurveyBanner.ts

```diff
@@ -149,12 +159,11 @@ export class DataScienceSurveyBanner implements IJupyterExtensionBanner {
     public async launchSurvey(): Promise<void> {
         this.browserService.launch(this.surveyLink);
     }
-
     private async disable(monthsTillNextPrompt: number) {
-        const expiration = monthsTillNextPrompt * 31 * 24 * 60 * 60 * 1000;
-        await this.persistentState
-            .createGlobalPersistentState<boolean>(DSSurveyStateKeys.ShowBanner, false, expiration)
-            .updateValue(false);
+        await this.showBannerState.updateValue({
+            expiry: monthsTillNextPrompt * 31 * MillisecondsInADay,
+            data: true
```

Comment:
I think we want false and not true here in disable.

Reply:
Doesn't get used, I'm looking at the `expiry` value instead.

Reply:
Basically if there is no `expiry` then its enabled, else its disabled based on expiry value.

Reply:
Ahh got it now, I see that in the enabled function implementation. Yeah, that works. I might toss a comment here at this point? Without looking at the enabled implementation it's a bit non-intuitive.

Reply:
@DonJayamanne Wait hang on one more second. Don't you need to add a Date.now() onto the value here? disable(1) will be the same value today and tomorrow. Persistent state expiry did that under the surface.

Reply:
Thanks

Reply:
Phew, big bug, thanks. Tests pass as its a relative time.

Reply:
Yeah, that's the tricky bit with tests like that. You don't want tests to wait a month so you gotta fake it up somehow :).